client/my-sites/plans-features-main/index.jsx

```diff
@@ -153,8 +153,14 @@ export class PlansFeaturesMain extends Component {
 			term = TERM_ANNUALLY;
 		}
 
+		const plansFromProps = planTypes
+			.map( type => findPlansKeys( { group, term, type } )[ 0 ] )
+			.filter( plan => plan );
+
```

Comment:
@jonathansadowski What do you think we should do in the case of an invalid plan type specified to the component?

Here I'm filtering them out and then falling back to default plans if no types specified by `planTypes` are valid. However I'm wondering what's better:
- Graceful fallback like this: makes sure users never see a empty plan page because of a runtime error, but hides potential mistakes in what plan types are specified
- Alternatively, intentionally throwing an error in this case to alert a developer to the invalid plan type

Thoughts?

Reply:
I'm also torn. Given how long something broken can stay undiscovered, even when it's obvious, I'm almost leaning toward the former. Something like this: If the filtered `plansFromProps` is empty, treat it as if `plansFromProps` hadn't been supplied at all, and maybe write some message indicating that happened with `debug`.

Reply:
Using `warn`, as discussed in slack.